### PR TITLE
fix: 15535688 - not able to change token in table

### PR DESCRIPTION
**Pull request by krawczykkatarzyna**
Single payment action:

not able to select token in the table.

### CODE REVIEW

**Review by KornelKwiatkowski — APPROVED** [review]

File: src/components/v5/common/ActionSidebar/partials/AmountField/types.ts [hunk 1]
```diff
@@ -6,4 +6,5 @@ export interface AmountFieldProps {
   name: string;
   amount?: string;
   defaultToken?: string;
+  tokenAddressDisabled?: boolean;
```
Comment (TomaszChmielPagepro) [inline src/components/v5/common/ActionSidebar/partials/AmountField/types.ts:9]: Instead I would just add an optional `tokenAddress` prop then if it's set I would just render this part
image
but without button.
Then in SimplePaymentForm in transform all transaction would just use token from `amount` field.

File: src/components/v5/common/ActionSidebar/partials/TransactionTable/TransactionTable.tsx [hunk 2]
```diff
@@ -64,7 +72,7 @@ const TransactionTable: FC<TransactionTableProps> = ({ name }) => {
             fieldArrayMethods.append({
               amount: {
                 amount: '0',
-                tokenAddress: nativeToken?.tokenAddress || '',
+                tokenAddress: tokenAddressController.value || '',
```
Comment (TomaszChmielPagepro) [inline src/components/v5/common/ActionSidebar/partials/TransactionTable/TransactionTable.tsx:75]: This line can be removed after implement this comment https://github.com/JoinColony/colonyCDapp/pull/1214/files#r1373098797

File: src/components/v5/common/ActionSidebar/partials/TransactionTable/TransactionTable.tsx [hunk 3]
```diff
@@ -35,6 +39,10 @@ const TransactionTable: FC<TransactionTableProps> = ({ name }) => {
   const { getFieldState } = useFormContext();
   const fieldState = getFieldState(name);
 
+  const { field: tokenAddressController } = useController({
+    name: 'amount.tokenAddress',
+  });
```
Comment (TomaszChmielPagepro) [inline src/components/v5/common/ActionSidebar/partials/TransactionTable/TransactionTable.tsx:44]: This shouldn't be here

File: src/components/v5/common/ActionSidebar/partials/AmountField/AmountField.tsx [hunk 4]
```diff
@@ -53,7 +53,7 @@ const AmountField: FC<AmountFieldProps> = ({ name }) => {
     onInput,
     formattingOptions,
     selectedToken,
-  } = useAmountField(tokenAddressController.value);
+  } = useAmountField(tokenAddressController.value || tokenAddress);
```
Comment (TomaszChmielPagepro) [inline src/components/v5/common/ActionSidebar/partials/AmountField/AmountField.tsx:56]: It should be the other way around. Prop overrides value

File: src/components/v5/common/ActionSidebar/partials/AmountField/AmountField.tsx [hunk 5]
```diff
@@ -99,93 +99,102 @@ const AmountField: FC<AmountFieldProps> = ({ name }) => {
...
-              )}
-              onClick={toggleTokenSelect}
-              aria-label={formatMessage({ id: 'ariaLabel.selectToken' })}
-            >
+          {tokenAddress ? (
+            <div className="flex items-center gap-2">
               <TokenIcon token={selectedToken || colonyTokens[0]} size="xs" />
               <span className="text-md">
                 {selectedToken?.symbol || colonyTokens[0].symbol}
               </span>
```
Comment (TomaszChmielPagepro) [inline src/components/v5/common/ActionSidebar/partials/AmountField/AmountField.tsx:107]: Seems like this has been reapeated 3 times across this component, can we create a variable for this content?

File: src/components/v5/common/ActionSidebar/partials/forms/SimplePaymentForm/SimplePaymentForm.tsx [hunk 6]
```diff
@@ -16,6 +17,7 @@ const displayName = 'v5.common.ActionSidebar.partials.SimplePaymentForm';
 
 const SimplePaymentForm: FC<ActionFormBaseProps> = ({ getFormOptions }) => {
   const intl = useIntl();
+  const tokenAddress = useWatch({ name: 'amount.tokenAddress' });
```
Comment (TomaszChmielPagepro) [inline src/components/v5/common/ActionSidebar/partials/forms/SimplePaymentForm/SimplePaymentForm.tsx:20]: can you move that to `useSimplePayment` ?

**Review by TomaszChmielPagepro — APPROVED** [review]

**Pull request merged**